torch/csrc/autograd/functions/pybind.h

```diff
@@ -11,21 +11,4 @@ namespace py = pybind11;
 
 namespace pybind11 { namespace detail {
 
-// handle Python <-> torch::autograd::Function conversions
-template <> struct type_caster<std::shared_ptr<torch::autograd::Function>> {
```

Comment:
Well, Python side `grad_fn` poking still works after this patch, so that didn't exercise this code. I think this specifically only affects people who were writing pybind11 bindings for autograd functions. I have the tiniest violin.